lib/reek/tree_dresser.rb

```diff
@@ -10,6 +10,47 @@ def initialize(klass_map = AST::ASTNodeClassMap.new)
       @klass_map = klass_map
     end
 
+    # Given this sexp (Parser::AST::Node):
+    # (class
+    #   (const nil :C) nil
+    #   (def :m
+    #     (args)
+    #     (send nil :puts
+    #       (str "nada"))))
+    #
+    # `dress` would now:
+    #
+    # 1.) determine the type of this sexp which would be `class`
+    # 2.) iterate over the children which would be:
+    #   [1] (const nil :C)
+    #   [2] nil
+    #   [3] (def :m ....
+    #
+    # Let's just look at what happens with [1]:
+    #   - `type` is going to be `const`
+    #   - children would be [nil, :C]
+    #   - the resulting class from @klass_map.klass_for(type)
+    #     would be an anonymous class with Reek::AST::ASTNode as superclass,
+    #     so not Parser::AST::Node anymore and it would have included
+    #     Reek::AST::SexpExtensions::ConstNode.
+    #     The overall ancestors chain just for [1] would look like this:
+    #     [ Reek::AST::SexpNode, Reek::AST::SexpExtensions::ConstNode, Reek::AST::ASTNode,
```

Comment:
I'm a little worried that all these details can become out-of-sync with reality very quickly. Can we explain all this in a more general way?